### PR TITLE
Added sub-page anchoring for all major elements

### PR DESCRIPTION
**Pull request by NewAlexandria**
### Replaces

https://github.com/mojombo/semver/pull/283
### Solves

https://github.com/mojombo/semver.org/issues/27
https://github.com/mojombo/semver/issues/222

@Haacked is this the correct place / format?

@adjohnson916 @tomByrer et. al.

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -16,12 +17,15 @@ Given a version number MAJOR.MINOR.PATCH, increment the:
...
+to the MAJOR.MINOR.PATCH format.
 
-Introduction
+<a name="intro"></a>
+# Introduction
 ------------
+<sup>[[link](#intro)]</sup>  
 
-In the world of software management there exists a dread place called
+In the world of software management there exists a dreaded place called
```
Comment (haacked) [inline index.md]: Could you remove any changes unrelated to adding anchors to this PR? The text should exactly match what's currently on http://semver.org.

**Comment by NewAlexandria** [comment]
@Haacked OK

**Comment by haacked** [comment]
Thanks! I'll take another look when I get a chance. :smile:

File: index.md [hunk 2]
```diff
@@ -3,10 +3,11 @@ layout: default
 title: Semantic Versioning 2.0.0
 ---
 
+
 Semantic Versioning 2.0.0
 ==============================
 
-Summary
```
Comment (haacked) [inline index.md:9]: Sorry if I wasn't clear before. I wanted these changes reverted too. I was hoping that this change would _only_ include the addition of named anchors. Changing the heading style here makes the headings all into `H1` elements which looks too large and gives these headings the same emphasis as the title.

**Comment by NewAlexandria** [comment]
I've removed the markdown formatting.  It seemed to dictate adding a couple other sub-page anchors, where the markdown headers would have made their own.  Hope this help but let me know; I'm glad the change is worthwhile and I'd just like to be done with it.

**Comment by haacked** [comment]
> Hope this help but let me know; I'm glad the change is worthwhile and I'd just like to be done with it.

I understand. Unfortunately, the links broke the markdown list item numbering so every item renders as 1. instead of the correct item number. I think we should probably make the number be the link to that section. And the headers link to their section. I can make this change for you if you'd rather be done with this. :smile:

**Comment by haacked** [comment]
I have a fix forthcoming.

**Comment by NewAlexandria** [comment]
Thanks @Haacked.  This week has a fair bit in it.  I would have gone back and made the change, but I probably wouldn't have taken the time to suss out [a rollover solution like you put.](http://semver.org/#spec-item-5)   It looks nice, and it's great to have links like this.  Thanks for publishing; glad to get the ball rolling.   :potable_water:

**Comment by haacked** [comment]
@NewAlexandria no problem. Without your effort, I wouldn't have figured this out! Turns out this is a big challenge in Jekyll. :smile: I'm going to blog about what I did and hopefully spur someone out there to submit a PR to Kramdown to add support for list item id generation.

And if nobody does, I guess I'll have to dust off my Ruby skills (what little I have) and see if I can do it myself. :smile:

**Comment by NewAlexandria** [comment]
Sounds good.  Glad to see you post the link here when you do, so that future maintenance 'strategy' is captured here via your article. :+1:

**Comment by haacked** [comment]
http://haacked.com/archive/2016/01/20/semver-deep-links/

**Pull request closed without merging**